src/main/java/com/yahoo/identity/services/session/SessionCreate.java

```diff
@@ -4,12 +4,24 @@
 
 public interface SessionCreate {
 
+    @Nonnull
+    String getUsername();
+
     @Nonnull
     SessionCreate setUsername(@Nonnull String username);
 
+    @Nonnull
+    String getPassword();
+
     @Nonnull
     SessionCreate setPassword(@Nonnull String password);
 
     @Nonnull
-    Session create();
+    SessionCreate setCredential(@Nonnull String credStr);
```

Comment:
Callers of `SessionCreate` shoudn't need to set the credential.
`SessionCreate` should take care of talking to the `CredentialService`.

Reply:
If I get it right, do you mean we use implicitly embed the `set credential process` into `create()` using `CredentialService`?